.github/workflows/test.yml

```
@@ -0,0 +1,49 @@
+name: test
+
+on:
+  # Run daily at midnight
+  schedule:
+    - cron: "0 0 * * *"
+  push:
+    branches: [ "main" ]
+    paths-ignore: [ '*.md', '*.asciidoc' ]
+  pull_request:
+    branches: [ "main" ]
+    paths-ignore: [ '*.md', '*.asciidoc' ]
+
+permissions:
+  contents: read
+
+jobs:
+
+  pre-commit:
+    if: github.event_name != 'schedule'
+    name: Run pre-commit
+    runs-on: ubuntu-latest
+    steps:
+    - uses: elastic/apm-pipeline-library/.github/actions/pre-commit@current
+
+  lint:
+    if: github.event_name != 'schedule'
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+      with:
+        python-version: '3.7'
+    - run: .ci/scripts/lint.sh
+
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python: [ '3.7', '3.8', '3.9', '3.10' ]
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
```

```suggestion
    - uses: actions/setup-python@v4
```
We could use the [v4](https://github.com/actions/setup-python/releases/tag/v4) instead of v3.